src/borg/archive.py

```diff
@@ -1387,21 +1389,29 @@ def process_item(self, archive, target, item):
 
     def process_chunks(self, archive, target, item):
         """Return new chunk ID list for 'item'."""
-        # TODO: support --compression-from
         if not self.recompress and not target.recreate_rechunkify:
             for chunk_id, size, csize in item.chunks:
                 self.cache.chunk_incref(chunk_id, target.stats)
             return item.chunks
         new_chunks = self.process_partial_chunks(target)
         chunk_iterator = self.create_chunk_iterator(archive, target, item)
         consume(chunk_iterator, len(new_chunks))
+        compress = self.compression_decider1.decide(item.path)
```

Comment:
I guess. Kinda depends on how the patterns exactly look. Btw. the patterns are matched to paths relative to how they are specified on the command line in borg-create?